opentracing-api/src/main/java/io/opentracing/Tracer.java

```diff
@@ -78,12 +78,14 @@
    * @param format the Format of the carrier
    * @param carrier the carrier for the SpanContext state. All Tracer.extract() implementations must support
    *                io.opentracing.propagation.TextMap and java.nio.ByteBuffer.
-   * @returns the SpanContext instance extracted from the carrier
+   *
+   * @return the SpanBuilder instance holding context to create a Span.
+   * A noop builder is returned if the carrier held no trace information.
    *
    * @see io.opentracing.propagation.Format
    * @see io.opentracing.propagation.Format.Builtin
    */
-  <C> SpanContext extract(Format<C> format, C carrier);
+  <C> SpanBuilder extract(Format<C> format, C carrier);
```

Comment:
What happens next to the SpanContext?
It is always, to my understanding, in some form or another, toward the creation of a Span.
Hence I made this change because I presumed that the only subsequent action was to then build a span.
Thanks for the use-case @yurishkuro, that was ofc my next question :-)

The counter use-case I have for this is creating the fluent API.
The extract method should return something that fluently follows for instrumentation code.
In this case it returns a NoopSpanBuider.

I'm not sure how a SpanContext alone can do that for you.
To solve @yurishkuro's use-case the SpanBuilder can extend SpanContext.
This prevents the waste that is described, as baggage can be taken from the builder without the creation of a Span. At the same time it provides the fluent api. 

These changes have been pushed.